images/base/entrypoint

```diff
@@ -80,11 +80,22 @@ fix_kmsg() {
   fi
 }
 
+configure_proxy() {
+  # ensure all processes receive the proxy settings by default
+  # https://www.freedesktop.org/software/systemd/man/systemd-system.conf.html
+  mkdir -p /etc/systemd/system.conf.d/
+  cat <<EOF >/etc/systemd/system.conf.d/proxy-default-environment.conf
+[Manager]
+DefaultEnvironment="HTTP_PROXY=${HTTP_PROXY:-}" "HTTPS_PROXY=${HTTPS_PROXY:-}" "NO_PROXY=${NO_PROXY:-}"
```

Comment:
Great way :+1: , so containerd and kubelet will be proxied, but I'm not sure if there will be other problems after kubelet is proxyed.

Reply:
Made a simple attempt, should we also add a `NO_PROXY` for local network?

```
172.17.0.3 - - [09/Jul/2019 11:03:28] "CONNECT 172.17.0.3:6443 HTTP/1.1" 501 -                                                                                                                                       
172.17.0.3 - - [09/Jul/2019 11:03:28] code 501, message Unsupported method ('CONNECT')                                                                                                                               
172.17.0.3 - - [09/Jul/2019 11:03:28] "CONNECT 172.17.0.3:6443 HTTP/1.1" 501 -                                                                                                                                       
172.17.0.3 - - [09/Jul/2019 11:03:28] code 501, message Unsupported method ('CONNECT')                                                                                                                               
172.17.0.3 - - [09/Jul/2019 11:03:28] "CONNECT 172.17.0.3:6443 HTTP/1.1" 501 -         
```

Reply:
Actually not necessary. :upside_down_face: 
https://github.com/kubernetes-sigs/kind/blob/99d872072e29a2ebdf0b4128743bf904b2ce1711/pkg/cluster/nodes/create.go#L157-L164

Reply:
Possible remaining problems:
- When we exec not under systemd (eg kubeadm)
- Certs

Reply:
kubeadm takes the proxy variables from the environment, but better confirm with @neolit123

Reply:
It does but they won't be supplied in the current implementation.
Pretty sure `docker exec` does not inherit the container's environment*

\* It does inherit ENV from a built image IIRC.

Reply:
Ideally we don't want kubeadm touching the internet anyhow, however this may affect other commands later.

Reply:
```
docker exec --privileged kind-worker2 env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=kind-worker2
NO_PROXY=172.17.0.0/16,2001:db8:1::/64,
no_proxy=172.17.0.0/16,2001:db8:1::/64,
HTTP_PROXY=http://192.168.2.188:3128
http_proxy=http://192.168.2.188:3128
```

Reply:
> Great way +1 , so containerd and kubelet will be proxied, but I'm not sure if there will be other problems after kubelet is proxyed.

Pods are still not going to get this I think. we could probably do something to make that happen but I'm not even sure if we should.